.github/workflows/rust.yml

```diff
@@ -102,16 +104,21 @@ jobs:
         uses: ./.github/actions/setup-builder
         with:
           rust-version: stable
+      - name: Check datafusion-substrait (default features)
+        run: cargo check --profile ci --all-targets -p datafusion-substrait
       - name: Check datafusion-substrait (no-default-features)
         run: cargo check --profile ci --all-targets --no-default-features -p datafusion-substrait
+        #
+        # Note: Only check libraries (not --all-targets) to cover end user APIs
```

Comment:
Rhe idea is to test only `--all-targets` for the default features (what a developer would likely run), otherwise the CI will just verify that a user of the crate can compile things rather than check all tests and benches too